### PR TITLE
Examples restructured and Modularized

### PR DESCRIPTION
**Pull request by alifarhid**
@hadia206 @ritwika314 @anudeep-bodo I revamped the entire Bodo-examples repo to be restructured in modules so that a novice user can follow and realize where to start and continue. 
Please review this as it is an important part of streamlining user experience. I believe the best way to review this is to pull and checkout the ex-restructure branch locally to be able to use Mac finder and VS code to visualize the new structure. 

Examples are not internally changed or tested, except deleting some redundant examples in snowflake and iceberg.

You may see some examples are repeated in folder `10-Bodo-Free-Trial` as they are small scale data but same code.

### CODE REVIEW

**Review by anudeep-bodo — APPROVED** [review]

File: 01-Basics-of-HPC-with-Python-SQL/environment.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Bodo
+channels:
+  - defaults
+  - conda-forge
+  - bodo.ai
+dependencies:
+  - python=3.9
```
Comment (hadia206) [inline 01-Basics-of-HPC-with-Python-SQL/environment.yaml:7]: why python=3.9 not python=3.10?

File: 02-Getting-Started-with-Bodo/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Bodo Tutorial
+Welcome to Bodo Tutorials!
+
+First make sure you have Bodo [installed](https://docs.bodo.ai/installation_and_setup/install/) by following the commands below.
+Bodo can be installed using Conda. To view tutorials with Jupyter Notebook, install `jupyter` and `ipyparallel` in the same enviroment where Bodo is installed.
+
+```shell
+conda create -n Bodo python=3.9 -c conda-forge
```
Comment (hadia206) [inline 02-Getting-Started-with-Bodo/README.md:8]: same why `3.9` ?
Also, why not use `pip`?

File: README.md [hunk 3]
```diff
@@ -1,54 +1,23 @@
...
-Many of the example scripts can take in optional arguments.
-`python <path>/<script.py> --help` shows the usage.
+## Modules outline
 
-By default all examples scripts can be run from the
-top directory (Bodo-examples) without any changes.
-Otherwise, make sure to change path of data files.
-For more information on the examples,
-please see the docstring at the top of each python script.
+You will first learn about the fundamentals of high performance computing using Python and Bodo in module 01.
```
Comment (hadia206) [inline README.md:16]: IMO It'll be better if you add link to module 1 here as well.

File: README.md [hunk 4]
```diff
@@ -1,54 +1,23 @@
...
 
-By default all examples scripts can be run from the
-top directory (Bodo-examples) without any changes.
-Otherwise, make sure to change path of data files.
-For more information on the examples,
-please see the docstring at the top of each python script.
+You will first learn about the fundamentals of high performance computing using Python and Bodo in module 01. 
 
-- [TPCH Queries](https://github.com/Bodo-inc/Bodo-examples/tree/master/examples/tpch), more information on TPC-H can be found [here](http://www.tpc.org/tpch/)
+Then you will learn more coding tricks and styles to write your bodo code in best way possible in module 02.
```
Comment (hadia206) [inline README.md]: Same a link to module 2

**Review by hadia206 — COMMENTED** [review]
This looks Great!
Thanks a lot @alifarhid 
I left small comments.
I didn't review the notebooks. Let me know if there's a specific notebook you want me to review.

File: README.md [hunk 5]
```diff
@@ -1,54 +1,23 @@
...
 
+We recommend you to run these examples on [Bodo Platform](https://platform.bodo.ai/account/sign-up). Just sign up for free to try. Some examples like modules 1 to 6 require a small clusters, e.g., 2 nodes of c5.2xlarge with total of 8 physical cores (16 vCPU) and 32GB RAM, and some examples need larger clusters. The description on top of each example explain what size of a cluster is required. For any size larger than 2 nodes c5.2xlarge, you can run at no Bodo charge within 14 days of sign up, but you or your organization would need to pay for AWS or Azur resources. After 14 days, you would incur bodo charges in addition to your cloud provider charges. But no worries, if you need more time, please reach us at [Contact Us](https://www.bodo.ai/contact) to extend that period for you.
```
Comment (ritwika314) [inline README.md:9]:
```suggestion
We recommend that you run these examples on the [Bodo Platform](https://platform.bodo.ai/account/sign-up). You can sign up for free to try it out. Some examples like modules 1 to 6 can run on small clusters, e.g., 2 nodes of c5.2xlarge with total of 8 physical cores (16 vCPU) and 32GB RAM, and some examples need larger clusters. The description provided with each example indicates the size of cluster that is required to run it. When you sign up for Bodo, we provide a hosted trial which lets you run the examples which only require the 2 node c5.2xlarge cluster for free, for 30 hours a month.  For any size larger than 2 nodes c5.2xlarge, you can use Bodo for free on your AWS or Azure account for 14 days after you sign up, and only pay your cloud provider for the resources used. Please [Contact Us](https://www.bodo.ai/contact) if you want a longer trial.

File: README.md [hunk 6]
```diff
@@ -1,54 +1,23 @@
...
 
+We recommend you to run these examples on [Bodo Platform](https://platform.bodo.ai/account/sign-up). Just sign up for free to try. Some examples like modules 1 to 6 require a small clusters, e.g., 2 nodes of c5.2xlarge with total of 8 physical cores (16 vCPU) and 32GB RAM, and some examples need larger clusters. The description on top of each example explain what size of a cluster is required. For any size larger than 2 nodes c5.2xlarge, you can run at no Bodo charge within 14 days of sign up, but you or your organization would need to pay for AWS or Azur resources. After 14 days, you would incur bodo charges in addition to your cloud provider charges. But no worries, if you need more time, please reach us at [Contact Us](https://www.bodo.ai/contact) to extend that period for you. 
 
-## Running Example Scripts
+You can also run these examples locally using Bodo Community Edition, but it is not recommended as you need to go through some [installation process](https://docs.bodo.ai/2022.7/installation_and_setup/install/#install) while everything is already ready for you at https://platform.bodo.ai.
```
Comment (ritwika314) [inline README.md:11]:
```suggestion
You can also run these examples locally by [installing bodo](https://docs.bodo.ai/latest/installation_and_setup/install/#install)  on your laptop. However, we strongly recommend using the Bodo Platform as it provides a notebook environment with all the code available and required packages already installed for you.

File: README.md [hunk 7]
```diff
@@ -1,54 +1,23 @@
...
 
-By default all examples scripts can be run from the
-top directory (Bodo-examples) without any changes.
-Otherwise, make sure to change path of data files.
-For more information on the examples,
-please see the docstring at the top of each python script.
+You will first learn about the fundamentals of high performance computing using Python and Bodo in module 01. 
 
-- [TPCH Queries](https://github.com/Bodo-inc/Bodo-examples/tree/master/examples/tpch), more information on TPC-H can be found [here](http://www.tpc.org/tpch/)
+Then you will learn more coding tricks and styles to write your bodo code in the best way possible in module 02.
```
Comment (ritwika314) [inline README.md:18]:
```suggestion
Then you will learn some best practices to write your Bodo code in module 02.

File: README.md [hunk 8]
```diff
@@ -1,54 +1,23 @@
...
-Otherwise, make sure to change path of data files.
-For more information on the examples,
-please see the docstring at the top of each python script.
+You will first learn about the fundamentals of high performance computing using Python and Bodo in module 01. 
 
-- [TPCH Queries](https://github.com/Bodo-inc/Bodo-examples/tree/master/examples/tpch), more information on TPC-H can be found [here](http://www.tpc.org/tpch/)
+Then you will learn more coding tricks and styles to write your bodo code in the best way possible in module 02. 
 
-- [Beer Reviews](examples/beer-reviews/beer-reviews.py)
+Modules 3 and 4 focus on Bodo's sweet spot, meaning ETL applications. You will find examples with operational databases like PostgreSQL, Oracle, MySQL, a data warehouse like Snowflake in module 04, and a data Lakehouse example with Iceberg in module 05.
```
Comment (ritwika314) [inline README.md:20]:
```suggestion
Modules 3 and 4 focus on Bodo's sweet spot, i.e. compute heavy data transformations, through ETL applications. You will find examples with operational databases like PostgreSQL, Oracle, MySQL, a data warehouse like Snowflake in module 04, and a data Lakehouse example with Iceberg in module 05.

File: README.md [hunk 9]
```diff
@@ -1,54 +1,23 @@
...
 
-- [TPCH Queries](https://github.com/Bodo-inc/Bodo-examples/tree/master/examples/tpch), more information on TPC-H can be found [here](http://www.tpc.org/tpch/)
+Then you will learn more coding tricks and styles to write your bodo code in the best way possible in module 02. 
 
-- [Beer Reviews](examples/beer-reviews/beer-reviews.py)
+Modules 3 and 4 focus on Bodo's sweet spot, meaning ETL applications. You will find examples with operational databases like PostgreSQL, Oracle, MySQL, a data warehouse like Snowflake in module 04, and a data Lakehouse example with Iceberg in module 05. 
 
-- [NYC Parking Tickets](examples/nyc-parking/nyc-parking.py)
+You can then run more use case examples in module 06. Module 7-9 contain larger scale examples with Machine Learning, Business use cases (financial, transportation, etc.). Module 09 demonstrates a head-to-head competition with Spark, Dask and Ray. And finally module 10 contains more advanced topics like deploying bodo applications in kubernetes, containers, streamlit, or on your cloud provider through Terraform (AWS and Azure).
```
Comment (ritwika314) [inline README.md:22]:
```suggestion
Module 06 contains some example use-cases. Module 7-9 contain larger scale examples with Machine Learning, Business use cases (financial, transportation, etc.). Module 09 demonstrates a head-to-head competition with Spark, Dask and Ray. And finally, module 10 contains more advanced topics like deploying Bodo applications in Kubernetes, containers, Streamlit, or on your cloud provider through Terraform (AWS and Azure).

**Review by ritwika314 — COMMENTED** [review]
Thank you so much for this @alifarhid , hugely improves usability.

**Review by ritwika314 — APPROVED** [review]
added a review with comments. thanks!

File: 02-Getting-Started-with-Bodo/README.md (hunk 2 above)
Comment (alifarhid) [reply 02-Getting-Started-with-Bodo/README.md:8]: This whole readme needs to be updated anyway. Readme's within each folder is not updated yet.

File: README.md (hunk 3 above)
Comment (alifarhid) [reply README.md:16]: done. Added link for all modules in the readme.

**Comment by alifarhid** [comment]
Thanks @ritwika314 @hadia206 and @anudeep-bodo. I incorporated your comments and added a new paragraph in Readme.md about `What if I wanted to test my code with my data?` . If you could please review that paragraph too and then I will merge. 
Thanks.

**Pull request merged**